component/api.jsonnet

```diff
@@ -0,0 +1,143 @@
+local kap = import 'lib/kapitan.libjsonnet';
+local kube = import 'lib/kube.libjsonnet';
+local inv = kap.inventory();
+// The hiera parameters for the component
+local params = inv.parameters.lieutenant;
+
+local role = std.parseJson(kap.yaml_load('lieutenant/manifests/api/role.yaml'));
+local service_account = std.parseJson(kap.yaml_load('lieutenant/manifests/api/service_account.yaml'));
+local role_binding = std.parseJson(kap.yaml_load('lieutenant/manifests/api/role_binding.yaml'));
+local deployment = std.parseJson(kap.yaml_load('lieutenant/manifests/api/deployment.yaml'));
+local service = std.parseJson(kap.yaml_load('lieutenant/manifests/api/service.yaml'));
+
+
+local ingress = kube.Ingress('lieutenant-api') {
```

Comment:
You probably want to have at least the option to enable TLS for the ingress. This needs roughly the following in the Ingress object (apart from cert-manager annotations):

```
spec:
  tls:
    - hosts:
        - <hostname>
      secretName: lieutenant-api-cert
```